smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -420,22 +433,57 @@ private void generateErrorDeserializer(GenerationContext context, StructureShape
         writer.write("");
     }
 
+    private boolean isStreamingUnion(GenerationContext context, OperationShape operation) {
+        if (operation.getOutput().isPresent()) {
+            ShapeId shapeId = operation.getOutput().get();
+            StructureShape outputShape = context.getModel().expectShape(shapeId).asStructureShape().get();
+
+            Collection<MemberShape> memberValues = outputShape.getAllMembers().values();
+            Optional<MemberShape> first = memberValues.stream().findFirst();
+            Set<UnionShape> unionShapesWithTrait =
+                    context.getModel().getUnionShapesWithTrait(StreamingTrait.class);
+            boolean hasStreamingUnion = first.isPresent()
+                    && memberValues.size() == 1
+                    && unionShapesWithTrait
+                    .stream()
+                    .map(Shape::getId)
+                    .collect(Collectors.toSet())
+                    .contains(first.get().getTarget());
```

Comment:
Maybe this implementation can be simplified with [EventStreamIndex](https://awslabs.github.io/smithy/javadoc/1.8.0/software/amazon/smithy/model/knowledge/EventStreamIndex.html)